### DISCUSSION TITLE
block only some sites but allow homepage

### DISCUSSION THREAD

**Discussion by littbarski** [proginosko/LeechBlockNG#512] [Q&A] [answered]
Hello, first of all: thank you very much, this is the only solution for specific tasks of blocking, while Google (incl. Family Link) and others just don't want to allow blocking in the best way they could.

I don't find any solution (neither with 2 blocks) for:
- block all websites
- allow one homepage
- block all other site of this website

like (not working):
*
+wanttobeallowed.com/
wanttobeallowed.com/butnotthis

**Comment by proginosko** [comment]
The simplest solution is to use two block sets:

Block Set 1 (blocks everything except one website):
`*`
`+wanttobeallowed.com`

Block Set 2 (blocks everything on website except home page):
`wanttobeallowed.com/*+`

**Comment by littbarski** [comment]
thank you for the answer, yes this works as a beginning, but I actually saw that I have to block the subsites specifically, as some others may be allowed though :). 
I tried a third block set to put the allowed subsites in, but did not work, so the best solution would be to:
- block all websites
- allow one `wanttobeallowed.com` homepage
- block specific one or two subsites `wanttobeallowed.com/butnotthis`

**Comment by proginosko** [comment]
I'm not sure exactly what you're aimed it. Is the idea that you want to block some subsites as well as the home page?

**Comment by littbarski** [comment]
thank you for taking the time, no, the homepage, so everything directly under the site (see my last answer) is allowed, as well as some subsites. I could easily block all website in the internet and just allow this website as an exception. What is difficult is just to block one or two subsites while allowing this domain in general as an exception.

**Comment by littbarski** [comment]
I saw in another Q&A here below a similar, but opposite example, so perhaps I would need this as negative version, an exception? 
`^(https?|file):\/+((www\.)?(reddit|discord)\.com(\/(r|message)\/(all|popular|inbox|portugal))?\/?)$`

like
```
*
+^(https?|file):\/+((www\.)?(allowedsite)\.com(\/(r|message)\/(notallowed1|notallowed2))?\/?)$
```

(?) would probably not need the "message" part :)

**Comment by proginosko** [comment]
Let me try to get this straight. :smile: You want to block:

1. www.allowedsite.com (homepage)
2. www.allowedsite.com/notallowed1
3. www.allowedsite.com/notallowed2
4. All other sites completely

So, for example, www.allowedsite.com/subsite would _not_ be blocked.

Is that right?

**Comment by littbarski** [comment]
no, if I could "choose" the code, I would code it like:
```
* [blocks all internet]
+allowedsite.com [this domain/homepage is allowed as an exception, including all not blocked subsites, so allowedsite.com/allowed is allowed]
allowedsite.com/notallowed [this subsite is not allowed]
```

To summarize I would like to block all internet, allow only one domain, and then block again from this domain one subsite (not: subdmomain).

**Reply by proginosko** [reply, marked as answer]
Well, that's easy to do:

Block Set 1:
`*`
`+wanttobeallowed.com`

Block Set 2:
`wanttobeallowed.com/notallowed1`
`wanttobeallowed.com/notallowed2`
Etc.

**Comment by littbarski** [comment]
yes, indeed, it works now. Thank you. 
Don't know why I could not solve it, probably I first had to learn that there should be two block sets, and then I perhaps just created one quickly without changing the days and hours. 
And in the opposite way (for the future), if I would like to block a domain, but only allow one subsite, then it would be like: 
```
*
+normallynotallowed.com/allowedsubsite
```
So that normallyallowed.com is blocked with the * but because of the exception, this subsite would be allowed, correct?
(as said this is only for the future, you gave me a great solution)

**Reply by proginosko** [reply]
Correct.

**Reply by littbarski** [reply]
thank you again, very helpful help and helpful piece of software!
